core-bundle/src/EventListener/DataContainer/PageSearchListener.php

```diff
@@ -41,10 +41,10 @@ public function onSaveAlias(string $value, DataContainer $dc): string
         return $value;
     }
 
-    #[AsCallback(table: 'tl_page', target: 'fields.noSearch.save')]
-    public function onSaveNoSearch(string $value, DataContainer $dc): string
+    #[AsCallback(table: 'tl_page', target: 'fields.searchIndexer.save')]
+    public function onSaveSearchIndexer(string $value, DataContainer $dc): string
     {
-        if (!$value || (bool) $value === (bool) ($dc->getCurrentRecord()['noSearch'] ?? false)) {
+        if ($value === ($dc->getCurrentRecord()['searchIndexer'] ?? null) || !str_starts_with($value, 'never')) {
```

Comment:
I think this is still not correct. We want to purge the search index, if the value changes, right?

Reply:
I think it needs to be:

```suggestion
        if (!$value || 'always_index' === $value || $value === ($dc->getCurrentRecord()['searchIndexer'] ?? null)) {
```

i.e. return if:

* The value is empty
* or if the value is `always_index`
* or if the value is the same as before.